SWDEV-552620 - Skip copying kernelNames in graph launch path

sourabhuday opened the pull request:
## Motivation
Problem: 

<img width="973" height="109" alt="image" src="https://github.com/user-attachments/assets/31c5c86c-f8ca-456d-85b4-2a3d77a6b85c" />

Results:

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->
<img width="352" height="139" alt="image" src="https://github.com/user-attachments/assets/90679d48-26c8-48cf-9bd8-c47a3ce849be" />


## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->
This avoids expensive string copying while keeping names accessible for profiling. Kernel names are now accessed via reference set - zero copy optmization. 
## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->
Used hipGraphTopology.cc test in hip-test
## Test Result

<!-- Briefly summarize test outcomes. -->
- First launch at 1000 nodes-  ~40.7% faster
- Repeat launches times does not grow linearly(O(n)) with increasing number of nodes and no variance
- Repeat launches on average improved - 10x
  
## Submission Checklist

- [X] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

sourabhuday committed:
SWDEV-552620 - Optimize graph repeatLaunches

saleelk commented on projects/clr/rocclr/device/rocm/rocvirtual.cpp:
```diff
@@ -1365,11 +1365,8 @@ bool VirtualGPU::dispatchAqlPacketBatch(const std::vector<uint8_t*>& packets,
   dispatchBlockingWait();
 
-  // Add all kernel names in bulk
-  vcmd->addKernelNames(kernelNames);
```
is this not a bug, you took this off entirely?

sourabhuday replied:
Fixed it. Please review again.

saleelk commented on projects/clr/rocclr/platform/command.hpp:
```diff
@@ -1383,29 +1383,35 @@ class AccumulateCommand : public Command {
-  void addKernelNames(const std::vector<std::string>& kernelNames) {
-    kernelNames_.insert(kernelNames_.end(), kernelNames.begin(), kernelNames.end());
+  //! Move kernel names into internal storage and set reference to it
+  void moveKernelNames(std::vector<std::string>&& kernelNames) {
```
I dont understand why you need move and setKernelNames, this is messy, to be clean one only needs references to kernel names , which are in codeobject. that way we dont copy  copy any string around.

sourabhuday replied:
Please review the latest commit. It attempts to eliminate all string copies and references kernel names directly everywhere possible.

sourabhuday committed:
SWDEV-552620 - Store pointers during graph capture and batching

- Changed PacketBatch to store pointers to kernel names instead of copies
- Updated GraphNode to reference persistent kernel names via pointer
- Modified AccumulateCommand to work with vector of pointers
- Updated all dispatch functions to accept pointer-based vectors
- Adapted changes to work with new multi-device graph structures

This eliminates string copying overhead during graph capture and launch.

sourabhuday committed:
Merge branch 'develop' into amd/dev/sbetiger/skip_add_kernel_name
